Snippets/CosmosDB/CosmosDB_2/Transactions.cs

```diff
@@ -0,0 +1,182 @@
+    {
+        #region ExtractPartitionKeyFromHeadersExtractor
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromHeaders<MyAppsCustomConfigurationHolder>((headers, config) => new ContainerInformation(headers["ContainerNameHeader"], new PartitionKeyPath(config.PartitionKeyPath)), appConfig);
+
+        #endregion
+    }
+
+    void ExtractPartitionKeyFromHeadersCustom(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractPartitionKeyFromHeadersCustom
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractPartitionKeyFromHeaders(new CustomPartitionKeyFromHeadersExtractor());
+
+        #endregion
+    }
+
+    void ExtractContainerInfoFromHeader(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractContainerInfoFromHeader
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromHeader("ContainerKey", headerValue => new ContainerInformation(headerValue, new PartitionKeyPath("/partitionKey")));
+
+        #endregion
+    }
+
+    void ExtractContainerInfoFromHeaders(PersistenceExtensions<CosmosPersistence> persistence, MyAppsCustomConfigurationHolder appConfig)
+    {
+        #region ExtractContainerInfoFromHeaders
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromHeaders<MyAppsCustomConfigurationHolder>((headers, config) => new ContainerInformation(headers["ContainerNameHeader"], new PartitionKeyPath(appConfig.PartitionKeyPath)), appConfig);
+
+        #endregion
+    }
+
+    void ExtractContainerInfoFromHeadersCustom(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractContainerInfoFromHeadersCustom
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromHeaders(new CustomContainerInformationFromHeadersExtractor());
+
+        #endregion
+    }
+
+    void ExtractPartitionKeyFromMessageExtractor(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractPartitionKeyFromMessageExtractor
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractPartitionKeyFromMessage<MyMessage>(message => new PartitionKey(message.ItemId));
+
+        #endregion
+    }
+
+    void ExtractPartitionKeyFromMessageCustom(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractPartitionKeyFromMessageCustom
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractPartitionKeyFromMessage<MyMessage>(message => new PartitionKey(message.ItemId));
```

Comment:
This snippet needs to be adjusted to use the custom provider

Reply:
Fixed